src/main/java/com/auth0/AuthenticationController.java

```diff
@@ -205,31 +221,33 @@ public void doNotSendTelemetry() {
      * depending on the chosen Response Type, to finally obtain a set of {@link Tokens}.
      *
      * @param request the received request to process.
+     * @param response the received response to process.
      * @return the Tokens obtained after the user authentication.
      * @throws InvalidRequestException       if the error is result of making an invalid authentication request.
      * @throws IdentityVerificationException if an error occurred while verifying the request tokens.
      */
-    public Tokens handle(HttpServletRequest request) throws IdentityVerificationException {
+    public Tokens handle(HttpServletRequest request, HttpServletResponse response) throws IdentityVerificationException {
         Validate.notNull(request);
 
-        return requestProcessor.process(request);
+        return requestProcessor.process(request, response);
     }
 
     /**
      * Pre builds an Auth0 Authorize Url with the given redirect URI using a random state and a random nonce if applicable.
      *
-     * @param request     the caller request. Used to keep the session context.
+     * @param request     the request.
+     * @param response    the response. Used to store auth-based cookies.
      * @param redirectUri the url to call back with the authentication result.
      * @return the authorize url builder to continue any further parameter customization.
      */
-    public AuthorizeUrl buildAuthorizeUrl(HttpServletRequest request, String redirectUri) {
-        Validate.notNull(request);
+    public AuthorizeUrl buildAuthorizeUrl(HttpServletRequest request, HttpServletResponse response, String redirectUri) {
```

Comment:
Maintaining `HttpServletRequest` parameter for potential future use without introducing another potentially breaking change.